### PR TITLE
Split arithmetic ops from routines/math.h

### LINKED ISSUE #7414: Split arithmetic routines

### PR DESCRIPTION
**Pull request by emcastillo**
This closes #7414 

math.h only has now, rounding, checking and activation related functions

### CODE REVIEW

**Comment by emcastillo** [comment]
Jenkins, test this please.

File: chainerx_cc/chainerx/cuda/cuda_device/arithmetic.cu [hunk 1]
```diff
@@ -14,7 +14,9 @@
 #include "chainerx/cuda/numeric.cuh"
 #include "chainerx/device.h"
 #include "chainerx/dtype.h"
+#include "chainerx/kernels/arithmetic.h"
 #include "chainerx/kernels/math.h"
+#include "chainerx/routines/arithmetic.h"
 #include "chainerx/routines/math.h"
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device/arithmetic.cu]: Maybe you can remove `{kernels,routines}/math.h`?

File: chainerx_cc/chainerx/native/native_device/pool.cc [hunk 2]
```diff
@@ -24,6 +25,7 @@
 #include "chainerx/native/kernel_regist.h"
 #include "chainerx/native/tensor_dot.h"
 #include "chainerx/numeric_limits.h"
+#include "chainerx/routines/arithmetic.h"
```
Comment (niboshi) [inline chainerx_cc/chainerx/native/native_device/pool.cc]: I guess `kernels/arithmetic.h` is for `Divide(AS)Kernsl`, but do you need `routines/arithmetic.h`?

File: chainerx_cc/chainerx/routines/connection.cc [hunk 3]
```diff
@@ -18,10 +18,12 @@
 #include "chainerx/error.h"
 #include "chainerx/graph.h"
 #include "chainerx/kernel_registry.h"
+#include "chainerx/kernels/arithmetic.h"
 #include "chainerx/kernels/connection.h"
 #include "chainerx/kernels/linalg.h"
 #include "chainerx/kernels/math.h"
 #include "chainerx/macro.h"
+#include "chainerx/routines/arithmetic.h"
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/connection.cc]: Ditto. Do you need `routines/arithmetic.h`?

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Hmm.

```
 [TIMEOUT] /chainer/tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py (60 seconds)
```

The above error persisted even after a retry.

**Comment by emcastillo** [comment]
I tried it in my development environment and the test completed without issues

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**